src/Middleware/LoginSessionMiddleware.php

```diff
@@ -42,10 +44,14 @@ public function process(HTTPRequest $request, callable $delegate)
                 return $delegate($request);
             }
 
-            // Update LastAccessed date and IP address
-            $loginSession->LastAccessed = DBDatetime::now()->Rfc2822();
-            $loginSession->IPAddress = $request->getIP();
-            $loginSession->write();
+            // Update LastAccessed date and IP address if > that threshold
+            $date = DBDatetime::now()->Rfc2822();
+            $threshold = LoginSession::getUpdateThreshold();
+            if (strtotime($date) > strtotime($loginSession->LastAccessed) + $threshold) {
+                $loginSession->LastAccessed = $date;
+                $loginSession->IPAddress = $request->getIP();
+                $loginSession->write();
+            }
```

Comment:
Now that we're starting to add some logic around this, I wonder if it might be more appropriate to move all of this into a method in `LoginSession` itself and just call that method here?

Reply:
yes originally i added one method to refactor setting the LastAccessed + IPAddress, but didn't want to add more stuff than necessary to not overcomplicate the PR.
That's something that can be easily refactored later on, but if that's the only thing that get in the way i'll happily add it